### ISSUE TITLE
Add documentation about --dart-define key

### ISSUE THREAD

**Issue by TatsuUkraine** [flutter/website#4215]
Flutter 1.17 added support for custom compile-time variables in flutter tools with `--dart-define`

Quite interesting feature not just in terms of defining compile time variables for Dart, but also in terms of alternative way of environment configuration for application instead of flavors.

In my personal opinion, it would be helpful for developers to know about this feature when they're choosing the way how to setup they're configuration for multiple environments or just define compile-time variables.

I wrote [small article](https://link.medium.com/wsmmbWCe66)  about this feature and how it can be used instead of flavors. If it helps, feel free to use it)

**Comment by sfshaza2** [comment]
Yes, this feature in 1.17 took me a bit by surprise! Have to figure out where to put it...

**Comment by Roshan13046** [comment]
> Yes, this feature in 1.17 took me a bit by surprise! Have to figure out where to put it...

can you please describe where to put the documentation about --dart-define key?

**Comment by sfshaza2** [comment]
@Roshan13046, we now have a [Flutter CLI page](https://flutter.dev/docs/reference/flutter-cli). It should go there.

**Comment by Roshan13046** [comment]
> Flutter 1.17 added support for custom compile-time variables in flutter tools with `--dart-define`
> 
> Quite interesting feature not just in terms of defining compile time variables for Dart, but also in terms of alternative way of environment configuration for application instead of flavors.
> 
> In my personal opinion, it would be helpful for developers to know about this feature when they're choosing the way how to setup they're configuration for multiple environments or just define compile-time variables.
> 
> I wrote [small article](https://link.medium.com/wsmmbWCe66) about this feature and how it can be used instead of flavors. If it helps, feel free to use it)

The article link is not working. Can you please share a correct link for reference about the topic dart-define key?

**Comment by TatsuUkraine** [comment]
@Roshan13046 strange, it works for me, here is a friendly [link from medium](https://itnext.io/flutter-1-17-no-more-flavors-no-more-ios-schemas-command-argument-that-solves-everything-8b145ed4285d?source=friends_link&sk=256607486556b496d27397d6c238f131) let me know if it works

**Comment by sfshaza2** [comment]
I found another copy on [binary.studio](https://binary-studio.com/2020/06/23/flutter-3/).

**Comment by TatsuUkraine** [comment]
@sfshaza2 it might be not up to date, I was making changes in the article on medium after 1.20 flutter release

**Comment by TatsuUkraine** [comment]
@sfshaza2 thanks for reminding me about it, btw) need to update it as well)

**Comment by Roshan13046** [comment]
> @sfshaza2 thanks for reminding me about it, btw) need to update it as well)



> @Roshan13046 strange, it works for me, here is a friendly [link from medium](https://itnext.io/flutter-1-17-no-more-flavors-no-more-ios-schemas-command-argument-that-solves-everything-8b145ed4285d?source=friends_link&sk=256607486556b496d27397d6c238f131) let me know if it works

Thanks! The link is working now.
Should I include another row in the table of [https://flutter.dev/docs/reference/flutter-cli] for --dart-define key or write a separate section for --dart-define key?

**Comment by TatsuUkraine** [comment]
hi, just curious, is there any update?) Flutter 3.7 recently introduced another way to pass compile-time variables, which probably also good to have in the docs

If it helps, I made [another article about it](https://medium.com/@tatsu.ukraine/flutter-3-7-and-a-new-way-of-defining-compile-time-variables-f63db8a4f6e2), feel free to use it if needed
